### ISSUE TITLE
Unexpected IPv4-mapped IPv6 address returned

### ISSUE THREAD

**Issue by Blankwonder** [EAimTY/tuic#94]
Hi. I'm writing a compatible TUIC client implementation. I found that when using UDP relay mode native, even if the client sends a datagram to an IPv4 address (8.8.8.8 in this case), an IPv4-mapped IPv6 address may return. 

I'm not sure if it's expected behavior. Since the original address is in IPv4, the underlying socket should be an AF_INET socket.

Here is the server log

```
[2022-10-18T07:19:56Z DEBUG] [106.120.31.188:51142] [authentication]
[2022-10-18T07:19:56Z DEBUG] [106.120.31.188:51142] [packet-from-native] [1] [8.8.8.8:53]
[2022-10-18T07:19:56Z INFO ] [106.120.31.188:51142] [associate] [1]
[2022-10-18T07:19:56Z DEBUG] [106.120.31.188:51142] [packet-to-native] [1] [[::ffff:8.8.8.8]:53]
[2022-10-18T07:19:56Z INFO ] [106.120.31.188:51142] [dissociate] [1]
```

**Comment by EAimTY** [comment]
This is a bug on the server side. It will be fixed in the version.

**Comment by EAimTY** [comment]
Should be fixed in [tuic-server-1.0.0-pre-alpha0](https://github.com/EAimTY/tuic/releases/tag/tuic-server-1.0.0-pre-alpha0)
